shared/api/network.go

```diff
@@ -58,3 +58,23 @@ type NetworkLease struct {
 	Address  string `json:"address" yaml:"address"`
 	Type     string `json:"type" yaml:"type"`
 }
+
+// NetworkStateStats represents statistics about an interface
+//
+// API extension: network_state
+type NetworkStateStats struct {
+	BytesTransmitted   int64 `json:"bytes_transmitted" yaml:"bytes_transmitted"`
+	BytesReceived      int64 `json:"bytes_received" yaml:"bytes_received"`
+	PacketsTransmitted int64 `json:"packets_transmitted" yaml:"packets_transmitted"`
+	PacketsReceived    int64 `json:"packets_received" yaml:"packets_received"`
+}
+
+// NetworkState represents the state of an interface
+//
+// API extension: network_state
+type NetworkState struct {
+	MTU       int               `json:"mtu" yaml:"mtu"`
+	Hwaddr    string            `json:"hwaddr" yaml:"hwaddr"`
+	Addresses []string          `json:"addresses" yaml:"addresses"`
```

Comment:
This should be using a struct a NetworkStateAddress struct that's similar to ContainerStateNetworkAddress:
```golang
type ContainerStateNetworkAddress struct {
    Family  string `json:"family" yaml:"family"`
    Address string `json:"address" yaml:"address"`
    Netmask string `json:"netmask" yaml:"netmask"`
    Scope   string `json:"scope" yaml:"scope"`
}
```